pkgs/applications/networking/browsers/firefox/wrapper.nix

```
@@ -102,15 +102,12 @@ let
 
       nameArray = builtins.map(a: a.name) (lib.optionals usesNixExtensions nixExtensions);
 
-      requiresSigning = browser ? MOZ_REQUIRE_SIGNING
-                     -> toString browser.MOZ_REQUIRE_SIGNING != "";
-
       # Check that every extension has a unqiue .name attribute
       # and an extid attribute
       extensions = if nameArray != (lib.unique nameArray) then
         throw "Firefox addon name needs to be unique"
-      else if requiresSigning && !lib.hasSuffix "esr" browser.name then
-        throw "Nix addons are only supported without signature enforcement (eg. Firefox ESR)"
+      else if browser.requireSigning && !lib.hasSuffix "esr" browser.name then
+        throw "Nix addons are only supported without signature enforcement (eg. LibreWolf or Firefox Developer Edition)"
```

```suggestion
      else if browser.requireSigning then
        throw "Nix addons are only supported without signature enforcement (eg. LibreWolf or Firefox Developer Edition)"
```

There is no ESR release which allows this anymore.

---

Hmm. The unfortunate consequence of another option is that (to my knowledge) nothing in Nixpkgs would work with that part of the wrapper by default, thus requiring compiling Firefox yourself.

Would it be reasonable/possible to add those variants to be built by hydra and cached?

---

Double checked LibreWolf and they allow addon sideloading by default, so that will at least be built by default. A version of Firefox Developer Edition compiled that allows addon sideloading would be nice, though.